teps/0044-data-locality-and-pod-overhead-in-pipelines.md

```diff
@@ -145,59 +145,170 @@ See [TEP-0074](./0074-deprecate-pipelineresources.md) for the deprecation plan f
 
 ## Requirements
 
-- Tasks can be composed and run together:
+1. Tasks can be composed and run together:
   - Must be able to share data without requiring a volume external to the pod
   - Must be possible to run multiple Tasks as one pod
-- It should be possible to have Tasks that run even if others fail; i.e. the Task
+1. It should be possible to have Tasks that run even if others fail; i.e. the Task
   can be run on the same pod as another Task that fails
-  - This is to support use cases such as uploading test results, even if the test
-    Task failed
-    - This requirement is being included because we could choose a solution that doesn't
-      address the above use case; for example in PipelineResources, you can have a
-      storage "output" but if the steps fail, the "output" pipelineresource will not run
-- Any configuration for the execution of a Pipeline must be modifiable at runtime.
+  - This is to support use cases such as uploading test results, even if the test Task failed
+  - This requirement is being included because we could choose a solution that doesn't address the above use case.
+1. Any configuration for the execution of a Pipeline must be modifiable at runtime.
```

Comment:
I'm missing some background here - what's the use case for modifying the execution config in a PipelineRun vs a Pipeline?

Reply:
it's very likely we will want a configuration option here on the Pipeline; I think this is probably something we should implement. The reason to allow people to specify this on a PipelineRun is generally because the choice of how to execute a Pipeline is a runtime level concern. Specifically, I can imagine situations where you might have different approaches to PVCs in different clusters but want to reuse the same Pipeline definition from a repo, or maybe some PipelineRuns will use different amounts of data than other PipelineRuns for the same Pipeline and you want to sometimes distribute the work to different pods and sometimes not. Basically-- I think we shouldn't lock ourselves into a solution that can only be specified at one of runtime or authoring time. (Perhaps this should read "must be modifiable at both runtime and authoring time, but the initial implementation will be runtime based"? This is the question I tried to address with #610.)

Reply:
That helps, thanks. I agree that we shouldn't lock ourselves into a solution that cannot be extended. 
The reason I was asking about having the config in a pipeline is that pipeline resources are specified in a pipeline/task spec. So, for the specific use case of replacing pipeline resources though (e.g. a git resource in a TaskRun), the decision of using a single pod/no PVCs was made by the task/pipeline author. 
It seems like the scope of this TEP is greater than just replacing the pipeline resources use case though it might make sense to do this at the Run